File: 4kyu/squared_spiral_3/complete_solution.py
import math


def squared_spiral(n):

    sqrt = math.floor(math.sqrt(n))
    remaining = n - (sqrt ** 2)

    if sqrt % 2 == 0:
        x = -int(sqrt / 2) + max(remaining - sqrt, 0)
        y = int(sqrt / 2) - min(remaining, sqrt)

    else:
        x = math.ceil(sqrt / 2) - max(remaining - sqrt, 0)
        y = -math.floor(sqrt / 2) + min(remaining, sqrt)

    return x, y

File: 5kyu/squared_spiral_1/javascript/complete_solution.js
const squaredSpiral = (n) => {
  let sqrt = Math.floor(Math.sqrt(n));
  let remaining = n - sqrt ** 2;

  let x;
  let y;

  if (sqrt % 2 == 0) {
    x = -Math.trunc(sqrt / 2) + Math.max(remaining - sqrt, 0);
    y = Math.trunc(sqrt / 2) - Math.min(remaining, sqrt);
  } else {
    x = Math.ceil(sqrt / 2) - Math.max(remaining - sqrt, 0);
    y = -Math.floor(sqrt / 2) + Math.min(remaining, sqrt);
  }

  return [x, y];
};
